wordpress-mobile/WordPress-Android
# [Jetpack focus] Remove carousel and implement WordPress landing screen redesign

Fixes #17072

### Description

This PR implements the WordPress landing screen redesign behind the respective feature flag. The approach is to create a revamped fragment to co-exist with the current implementation, and use this new fragment conditionally (based on the flag) from the login activity. When the feature flag is turned on by default, the older flow can be removed.

This design is implemented via Jetpack Compose, ~~and a few things are not complete~~. To keep scope minimal for each PR, we can leverage the feature flag to develop partially complete UI with more granular feedback.

Since toggling the feature flag requires login, and navigating to this screen requires being logged out, it can be cumbersome to go back and forth between implementations. To make this easier, it is useful to apply the following patch:

<details>
<summary>Hard-coded flag patch</summary>

```diff
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java b/WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java
index df7d849e4d..fccaf5aa14 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java
@@ -227,7 +227,7 @@ public class LoginActivity extends LocaleAwareActivity implements ConnectionCall
     }
 
     private void loginFromPrologue() {
-        if (mLandingScreenRevampFeatureConfig.isEnabled()) {
+        if (true) {
             showFragment(new LoginPrologueRevampedFragment(), LoginPrologueRevampedFragment.TAG);
         } else {
             showFragment(new LoginPrologueFragment(), LoginPrologueFragment.TAG);
```
</details>

<details>

<summary>Former Solved Details</summary>

~~I also wired up one of the buttons to the login flow, because without the above hack, it isn't possible to toggle the feature flag off once logged out otherwise. The button functionality is scoped for a later PR, so this is an exception for practicality.~~

**Update:** Both login buttons were updated to have the expected functionality.

#### Questions

While this PR is still in draft, there are a couple of questions worth discussing before finalizing it:

1. The added font resource contributes significant size to the apk (~744k), but we only use this font in this one screen. Do we plan to use this font else-where, or is this considered acceptable?
    * There is a possibility to use "Downloadable fonts" for this, reducing the apk size, but this raises other questions, such as how to support users on older Android versions where this feature is not supported, etc.
2. The brush stroke image was rasterized directly from Figma as a 4x png. This seems to look ok in my testing, but it would be great to see how it looks on various form-factors. In the designs, the brush stroke is on the right side of the screen, but in this implementation, for RTL locales, it is on the left side of the screen. Is this ok?
    * Also, the image in RTL is not horizontally flipped. Since it is just a vertical brush stroke, this seemed ok to me, but it would be great to hear some other opinions.


**Note:** ~~The text for these buttons is based on future designs, and may possibly be reverted for the first iteration. I have deferred this decision for now.~~

Update: This has been reverted back to the former texts.

</details>

### Screenshots - Login Prologue

| Light | Dark |
|-|-|
|  ![wordpress-prologue](https://user-images.githubusercontent.com/8507675/187534368-a95ee014-ca4b-40f5-8727-4d08b52b6061.png) | ![wordpress-prologue-dark](https://user-images.githubusercontent.com/8507675/187534410-bd053fea-833d-42de-9ded-e433f0b67a83.png) |

| RTL | Landscape |
|-|-|
| ![wordpress-prologue-rtl](https://user-images.githubusercontent.com/8507675/187534656-e2a0b595-7585-4ebb-a864-501c1543d650.png) | ![wordpress-prologue-landscape](https://user-images.githubusercontent.com/8507675/187534674-91cc6a8c-b8d0-42e0-9e9b-a19946d5230a.png) |

To test:

Precondition: Enable the feature flag (either hard-coding via the patch above, or toggling within the app - or both)

1. Logout of the app once the feature flag is enabled
2. The new login screen should be visible
3. Put the device in dark mode
4. The dark mode design should be readable
5. Change to an RTL locale
6. Change the device orientation
7. Expect each view to be show as above

Login:

1. From the login screen, tap the "Log in or sign up with WordPress.com" button
2. Expect to be presented with the login flow
3. Go back
4. Tap the "Enter your existing site address" button
5. Expect to be presented with the site address login flow

Precondition: Disable the feature flag

1. Logout of the app
2. The old screen should be visible

Jetpack:

The Jetpack flavor of the app should be unaffected by these changes.

1. Build (or install from this PR) the Jetpack flavor of the app
2. Logout (with the feature enabled or disabled - it shouldn't matter)
6. The login screen should be unaffected

## Regression Notes
1. Potential unintended areas of impact
WordPress and Jetpack login screen

8. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

9. What automated tests I added (or what prevented me from doing so)
This PR only adds UI changes (under a feature flag).

PR submission checklist:
